### ISSUE TITLE
TypeError: Cannot read property 'request' of undefined

### ISSUE THREAD

**Issue by wholegroup** [olalonde/follow-redirects#92]
I use `axios 0.18.0` that `uses follow-redirects 1.5.5`. Sometimes I get this exception:

```
/home/andrey/projects/com-scanner/scanner/node_modules/follow-redirects/index.js:142
        nativeProtocol.request(this._options, this._onNativeResponse);
                       ^

TypeError: Cannot read property 'request' of undefined
    at RedirectableRequest._performRequest (/home/andrey/projects/com-scanner/scanner/node_modules/follow-redirects/index.js:142:24)
    at RedirectableRequest._processResponse (/home/andrey/projects/com-scanner/scanner/node_modules/follow-redirects/index.js:233:10)
    at ClientRequest.RedirectableRequest._onNativeResponse (/home/andrey/projects/com-scanner/scanner/node_modules/follow-redirects/index.js:39:10)
    at Object.onceWrapper (events.js:315:30)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:551:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:115:23)
    at Socket.socketOnData (_http_client.js:440:20)
    at emitOne (events.js:116:13)
```

**Comment by RubenVerborgh** [comment]
> Sometimes

Would you happen to know (or be able to narrow down) when?
It could be helpful is you could edit the index.js file to add `console.log(protocol)` right above and tell me the output.

I'm currently in the dark as to how this could occur, given that we explicitly assert this value to be valid: https://github.com/olalonde/follow-redirects/blob/v1.5.5/index.js#L278.

**Comment by wholegroup** [comment]
@RubenVerborgh, 

the output:
```
about: [ 'http:', 'https:' ]
```

and the url:
```
andrey@minidron:~$curl -v http://nancyleestewartartist.com/failed_request
*   Trying 192.241.119.19...
* TCP_NODELAY set
* Connected to nancyleestewartartist.com (192.241.119.19) port 80 (#0)
> GET /failed_request HTTP/1.1
> Host: nancyleestewartartist.com
> User-Agent: curl/7.61.0
> Accept: */*
> 
< HTTP/1.1 302 Object moved
< Cache-Control: private
< Content-Length: 132
< Content-Type: text/html
< Location: about:blank
< Server: Microsoft-IIS/7.5
< Set-Cookie: ASPSESSIONIDCQRADQTC=HOCELLKAEIMEFHJMINIBCBNH; path=/
< X-Powered-By: ASP.NET
< Date: Sun, 19 Aug 2018 19:45:50 GMT
< 
<head><title>Object moved</title></head>
<body><h1>Object Moved</h1>This object may be found <a HREF="about:blank">here</a>.</body>
* Connection #0 to host nancyleestewartartist.com left intact
```

**Comment by RubenVerborgh** [comment]
> about: [ 'http:', 'https:' ]

I'm very confused about where the `about` comes from.
Would you have any idea?

To increase clarity, could you please:

```JavaScript
console.log({ protocol: protocol, keys: Object.keys(this._options.nativeProtocols) })
```

**Comment by wholegroup** [comment]
@RubenVerborgh,

How to reproduce the bug
```js
require('follow-redirects').http.get('http://nancyleestewartartist.com/failed_request')
```

The output of `console.log`:
```
{ protocol: 'http:', keys: [ 'http:', 'https:' ] }
{ protocol: 'about:', keys: [ 'http:', 'https:' ] }
```

The url `http://nancyleestewartartist.com/failed_request` returns `Location: about:blank` as you can see from the `curl` output [above](https://github.com/olalonde/follow-redirects/issues/92#issuecomment-414151396).

**Comment by RubenVerborgh** [comment]
Right, now I fully get it. Will fix.

**Comment by RubenVerborgh** [comment]
The way I see it, this particular case will result in an `error` event with an error along the lines of `No handler for about: protocol`. Would that be fine, @wholegroup?

**Comment by wholegroup** [comment]
Yes, an error event will be fine!

**Comment by RubenVerborgh** [comment]
Published as v1.5.6, thanks for your help!
